src/redis/driver_impl.go

```
@@ -112,6 +112,24 @@ func NewClientImpl(scope stats.Scope, useTls bool, auth, redisSocketType, redisT
 	}
 	logger.Debugf("Implicit pipelining enabled: %v", implicitPipelining)
 
+	// Configure pool on-empty behavior to prevent connection storms during Redis failures
+	switch strings.ToUpper(poolOnEmptyBehavior) {
+	case "WAIT":
+		opts = append(opts, radix.PoolOnEmptyWait())
+		logger.Warnf("Redis pool %s: on-empty=WAIT (block until connection available)", maskedUrl)
+	case "CREATE":
+		opts = append(opts, radix.PoolOnEmptyCreateAfter(poolOnEmptyWaitDuration))
+		logger.Warnf("Redis pool %s: on-empty=CREATE after %v", maskedUrl, poolOnEmptyWaitDuration)
+	case "ERROR":
+		opts = append(opts, radix.PoolOnEmptyErrAfter(poolOnEmptyWaitDuration))
+		logger.Warnf("Redis pool %s: on-empty=ERROR after %v (fail-fast)", maskedUrl, poolOnEmptyWaitDuration)
+	default:
+		// Empty string = use radix default (PoolOnEmptyCreateAfter(1s))
+		if poolOnEmptyBehavior != "" {
+			logger.Warnf("Redis pool %s: unknown on-empty behavior '%s', using default (CREATE after 1s)", maskedUrl, poolOnEmptyBehavior)
+		}
```

Can we make the default behavior is CREATE and default wait time is 1s?

---

It will change the current default behavior. What do you think, @collin-lee?

---

What do you think, @collin-lee? Should we set the default value for REDIS_POOL_ON_EMPTY_BEHAVIOR to CREATE and REDIS_POOL_ON_EMPTY_WAIT_DURATION to 1s?

---

Isn't it the default behavior when poolOnEmptyBehavior is empty?

// Empty string = use radix default (PoolOnEmptyCreateAfter(1s))

I'm not sure, but my opinion is that I think it's better to fail fast in production.  What would be the case to set to CREATE/1s?

---

CREATE is the default to maintain backward compatibility with [radix's default behavior](https://github.com/mediocregopher/radix/blob/v3.8.1/pool.go#L325); ERROR is recommended for production, but changing the default would be a breaking change for existing deployments.